pyimagesearch.py

```diff
@@ -1,151 +1,167 @@
-            cv2.putText(orig, text, (startX, y),
-                        cv2.FONT_HERSHEY_SIMPLEX, 0.5, (0, 255, 0), 1)
-            elapsed_time = time.time() - starting_time
-            fps = frame_id / elapsed_time
-            fps_text = "FPS: " + str(round(fps, 2))
-            cv2.putText(orig, fps_text, (10, 10),
-                        cv2.FONT_HERSHEY_SIMPLEX, 0.5, (0, 255, 255), 1)
-        else:
-            elapsed_time = time.time() - starting_time
-            print("fps = " + str((round(frame_id / elapsed_time),2))
-    # show the output frame and wait for a key press
-    cv2.imshow("Frame", orig)
-    key = cv2.waitKey(1) & 0xFF
-    if key == 27 or key == 'q':
-        break
-# do a bit of cleanup
-print('cleanup')
-GPIO.cleanup()
-cv2.destroyAllWindows()
-video_stream.stop()
+
+def green_on():
+    light(GREEN_PIN, True)
+    light(YELLOW_PIN, False)
+
+
+def yellow_on():
+    light(GREEN_PIN, False)
+    light(YELLOW_PIN, True)
+
+
+def all_off():
+    light(GREEN_PIN, False)
+    light(YELLOW_PIN, False)
+
+
+def init_leds():
+    yellow_on()
+    time.sleep(1.0)
+    green_on()
+    time.sleep(0.5)
+    light(GREEN_PIN, False)
+
+
+def draw_box_and_labels(frame, box, label_text, fps_text):
+    (startX, startY, endX, endY) = box
+    cv2.rectangle(frame, (startX, startY), (endX, endY), (0, 255, 0), 2)
+    y = startY - 15 if startY - 15 > 15 else startY + 15
+    cv2.putText(frame, label_text, (startX, y),
+                cv2.FONT_HERSHEY_SIMPLEX, 0.5, (0, 255, 0), 1)
+    cv2.putText(frame, fps_text, (10, 10),
+                cv2.FONT_HERSHEY_SIMPLEX, 0.5, (0, 255, 255), 1)
+
+
+def main():
+    # Construct the argument parser and parse the arguments.
+    ap = argparse.ArgumentParser()
+    ap.add_argument("-m", "--model", default="mobilenet_ssd_v2_coco_quant_postprocess_edgetpu.tflite",
+                    help="path to TensorFlow Lite object detection model")
+    ap.add_argument("-l", "--labels", default="coco_labels.txt",
+                    help="path to labels file")
+    ap.add_argument("-c", "--confidence", type=float, default=0.3,
+                    help="minimum probability to filter weak detections")
+    ap.add_argument("-o", "--objects",
+                    default={0}, help="objects needed, default is 0 (person)")
+    ap.add_argument("-d", "--display", default=True,
+                    help="if false, there will be no display output")
+
+    # TODO: Parse pins as arguments
+    args = vars(ap.parse_args())
+
+    # Initialize the labels dictionary.
+    logging.info("Parsing class labels...")
+    labels = {}
+    # Loop over the class labels file.
+    for row in open(args["labels"]):
+        # Unpack the row and update the labels dictionary.
+        (classID, label) = row.strip().split(maxsplit=1)
+        labels[int(classID)] = label.strip()
+    # Load the Google Coral object detection model.
+    logging.info("Loading Coral model...")
+    model = DetectionEngine(args["model"])
```

Comment:
You should initialize all the args in the beginning, so you don't access args["some_parameter"] all the time.